src/player.cpp

```diff
@@ -11642,6 +11643,48 @@ std::vector<const item *> player::all_items_with_flag( const std::string &flag )
     } );
 }
 
+item player::item_with_best_of_quality( quality_id qid )
```

Comment:
```suggestion
item &player::item_with_best_of_quality( const quality_id &qid )
```
No need to copy the item. Nor is copying the quality id needed.